samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -121,14 +121,20 @@
   void sink(SinkFunction<? super M> sinkFn);
 
   /**
-   * Allows sending messages in this {@link MessageStream} to an {@link OutputStream}.
+   * Allows sending messages in this {@link MessageStream} to an {@link OutputStream} and then propogates the this
+   * {@link MessageStream} to the next chained operator
    * <p>
    * When sending messages to an {@code OutputStream<KV<K, V>>}, messages are partitioned using their serialized key.
    * When sending messages to any other {@code OutputStream<M>}, messages are partitioned using a null partition key.
-   *
+   * <p>
+   *  Note: There is no guarantee that message is first written to your configured output system and then only it is
```

Comment:
"Note: The message will be written but not flushed to the underlying output system before its propagated to the chained operators."

Reply:
Do you need to mention that the stream retains the original partitioning?

Reply:
Will do